### PR TITLE
Fix for camera look behavior getting stuck on

### LINKED ISSUE #5748: Cursor movement is replaced by camera rotation after specific interaction with entity in Viewport
**Describe the bug**
When clicking on the entity’s gizmo or arrows and holding it with the left mouse button while clicking the right mouse button, the cursor disappears and mouse movement is switched to camera rotation. Input can be changed back by clicking the right mouse button and moving the mouse while it is pressed.
 Please refer to the attached video for more information.
 
**Steps to reproduce**
1. Create an entity.
2. Press and hold the left mouse button when cursor is on entity’s gizmo.
3. Press the right mouse button while holding left mouse button and move the mouse.
4. Release the left mouse button.
5. Notice that the cursor disappears, and camera rotation is the default behavior of mouse movement.

**Expected behavior**
When the right mouse button is held during other interaction with an entity, the mouse movement is translated into the camera rotation until the right mouse button is released.

**Actual behavior**
When the right mouse button is pressed while entity’s gizmo is held with the left mouse button, the camera rotation becomes the default behavior of the mouse movement.

**Video**

https://user-images.githubusercontent.com/86953282/142438314-009487b4-58be-45eb-8309-a64e2df49ef3.mp4

**Found in Branch**
Stabilization_2110 (d222aee)

**Desktop/Device:**
 • Device: PC
 • OS: Windows
 • Version: 10 Pro
 • CPU: Intel i7 8700k
• GPU: Nvidia RTX 2080
 • Memory: 16GB

### PR DESCRIPTION
**Pull request by hultonha**
## What does this PR do?

Fixes #5748

Fixes irritating issue with camera when pressing RMB while interacting with a manipulator after having pressed RMB.

> Note: The changes to `ImmediateModeActionDispatcher` were needed to bring the behavior of the test code more inline with the Editor code. The logic more closely matches `ViewportManipulatorController.cpp` (see line 160 in that file).

```
MouseInteraction mouseInteraction = m_mouseInteraction;
if (overrideButton)
{
    mouseInteraction.m_mouseButtons.m_mouseButtons = static_cast<AZ::u32>(overrideButton.value());
}
```

## How was this PR tested?

1. Followed reproduction steps in #5748 and verified fix
2. Created an integration test to verify the behavior before/after

### Before

```bash
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ManipulatorCoreInteractionFixture
[ RUN      ] ManipulatorCoreInteractionFixture.MouseUpOfOtherMouseButtonDoesNotEndManipulatorInteraction
C:/dev/o3de/Code/Framework/AzToolsFramework/Tests/ManipulatorCoreTests.cpp(260): error: Value of: interacting
Expected: is true
  Actual: false (of type bool)
[  FAILED  ] ManipulatorCoreInteractionFixture.MouseUpOfOtherMouseButtonDoesNotEndManipulatorInteraction (8 ms)
[----------] 1 test from ManipulatorCoreInteractionFixture (8 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (10 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ManipulatorCoreInteractionFixture.MouseUpOfOtherMouseButtonDoesNotEndManipulatorInteraction
```

### After

```bash
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ManipulatorCoreInteractionFixture
[ RUN      ] ManipulatorCoreInteractionFixture.MouseUpOfOtherMouseButtonDoesNotEndManipulatorInteraction
[       OK ] ManipulatorCoreInteractionFixture.MouseUpOfOtherMouseButtonDoesNotEndManipulatorInteraction (7 ms)
[----------] 1 test from ManipulatorCoreInteractionFixture (9 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (10 ms total)
[  PASSED  ] 1 test.
```

### CODE REVIEW

File: Code/Framework/AzManipulatorTestFramework/Source/ImmediateModeActionDispatcher.cpp [hunk 1]
```diff
@@ -75,15 +75,19 @@ namespace AzManipulatorTestFramework
     {
         ToggleOn(GetMouseInteractionEvent()->m_mouseInteraction.m_mouseButtons.m_mouseButtons, MouseButton::Left);
         GetMouseInteractionEvent()->m_mouseEvent = AzToolsFramework::ViewportInteraction::MouseEvent::Down;
-        m_manipulatorViewportInteraction.GetManipulatorManager().ConsumeMouseInteractionEvent(*m_event);
+        auto mouseEvent = *GetMouseInteractionEvent();
+        mouseEvent.m_mouseInteraction.m_mouseButtons.m_mouseButtons = static_cast<AZ::u32>(MouseButton::Left);
```
Comment (cgalvan) [inline Code/Framework/AzManipulatorTestFramework/Source/ImmediateModeActionDispatcher.cpp:79]: Should we prefer `aznumeric_cast` on these?

**Review by cgalvan — APPROVED** [review]
Thanks for fixing this!

File: Code/Framework/AzManipulatorTestFramework/Source/ImmediateModeActionDispatcher.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzManipulatorTestFramework/Source/ImmediateModeActionDispatcher.cpp:79]: Ah yes good point I can update that :+1:

**Review by jonawals — COMMENTED** [review]
LGTM

**Review by gadams3 — APPROVED** [review]

**Pull request merged**